### PR TITLE
52/client message passing

### PR DESCRIPTION
**Pull request by alex-eg**

### CODE REVIEW

**Comment by Hairash** [comment]
#52

File: backend/src/websocket.rs [hunk 1]
```diff
@@ -1,20 +1,65 @@
...
+            c.do_send(Msg(m.clone()));
+        }
+    }
+
+    fn send_turn(&self) {
+        let message = api::common::Message::new(api::response::CMD_TURN);
+        let clients = self.app_state.clients.lock().unwrap();
+        debug!("Sending turn to other player");
+        for c in &*clients {
+            if *c == *self.self_addr.as_ref().unwrap() {
```
Comment (bronvic) [inline backend/src/websocket.rs:41]: reverting this condition will simplify a lot

File: backend/src/main.rs [hunk 2]
```diff
@@ -25,15 +29,34 @@ extern crate lazy_static;
...
+    let res = ws::start(
+        websocket::Websocket {
+            self_addr: None,
+            app_state: data.clone(),
+        },
+        &req,
+        stream,
+    );
+
+    let cnt = data.clients.lock().unwrap();
```
Comment (bronvic) [inline backend/src/main.rs:46]: this is not interesting debug - lets remove it

File: backend/src/websocket.rs [hunk 3]
```diff
@@ -1,20 +1,65 @@
...
-pub struct Websocket;
+#[derive(Debug)]
+pub struct Websocket {
+    pub self_addr: Option<Addr<Websocket>>,
+    pub app_state: web::Data<AppState>,
+}
+
+#[derive(Message, Debug)]
+#[rtype(result = "()")]
+pub struct Msg(pub String);
```
Comment (bronvic) [inline backend/src/websocket.rs:24]: let's think a bit about naming here

and maybe we can hide this into structs from api/response

File: backend/src/websocket.rs [hunk 4]
```diff
@@ -68,9 +120,15 @@ impl StreamHandler<Result<ws::Message, ws::ProtocolError>> for Websocket {
             Err(error) => ctx.text(error),
         }
 
-        ctx.text(&serde_json::to_string(&game).unwrap());
-        let message = api::common::Message::new(api::response::CMD_TURN);
-        ctx.text(&serde_json::to_string(&message).unwrap())
+        if clients_num == 2 {
+            debug!("Two clients connected, sending game!");
+            self.broadcast(game);
+            let message = api::common::Message::new(api::response::CMD_TURN);
```
Comment (bronvic) [inline backend/src/websocket.rs:126]: self.send_turn()

**Review by bronvic — COMMENTED** [review]
just commenting because i, maybe, would not have time to approve

**Review by bronvic — APPROVED** [review]
now it's fine

File: backend/src/websocket.rs (hunk 3 above)
Comment (Hairash) [reply backend/src/websocket.rs:24]: https://github.com/Miol-Mor/battle-game/issues/63

**Pull request merged**